Bam92/teamwork
# #168784882 add some more optional features

#### What does this PR do?
Have these features working: `filter articles by categories` and `flag articles or comments`
* `POST '/articles/:id/flag`
* GET /articles/categories/:id
* `POST /comments/:id/flag`
#### Description of Task to be completed?
- [x] Refactor the code base: add `data` directory and some new other files
- [x] Add new routes 
- [x] Fix unit test break
#### How should this be manually tested?
- Clone the repo
- `yarn install`
- `yarn start`
Use `Postman` or `curl` to test endpoints or `yarn test` to see my tests running
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#168784882](https://www.pivotaltracker.com/story/show/168784882)
